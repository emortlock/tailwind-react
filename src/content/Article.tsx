import type { ContentBlockProps } from './ContentBlock'

import React from 'react'

import { ContentBlock } from './ContentBlock'

export const Article = ({ as = 'article', ...rest }: ContentBlockProps) => (
  <ContentBlock as={as} {...rest} />
)
