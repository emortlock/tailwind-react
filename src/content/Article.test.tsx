import React from 'react'
import { render, RenderResult } from '@testing-library/react'

import { Article } from './Article'

const setup = (
  testProps = {},
): [RenderResult, { props: Record<string, any> }] => {
  const props = Object.assign(testProps)

  const wrapper = render(<Article {...props}>Lorem ipsum...</Article>)

  return [
    wrapper,
    {
      props,
    },
  ]
}

describe('Article', () => {
  it('renders matching snapshot', () => {
    const [{ asFragment }] = setup()

    expect(asFragment()).toMatchSnapshot()
  })
})
