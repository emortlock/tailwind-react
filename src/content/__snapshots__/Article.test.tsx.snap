// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders matching snapshot 1`] = `
<DocumentFragment>
  <article
    aria-labelledby="content-1"
    class="p-4"
  >
    Lorem ipsum...
  </article>
</DocumentFragment>
`;
